OsmIntegrator/Controllers/TileController.cs

```diff
@@ -254,59 +254,105 @@ public async Task<ActionResult<string>> RemoveUser(string id)
-                DbTile currentTile =
-                    await _dbContext.Tiles.Include(
-                        tile => tile.Users).SingleOrDefaultAsync(x => x.Id == Guid.Parse(id));
+                                   
+                    currentTile.Users.Clear();
+                    currentTile.Users.Add(user);
 
-                if (currentTile == null)
+                    _dbContext.SaveChanges();
+
+                    return Ok("Tile approved");
+                } catch (BadHttpRequestException e)
                 {
-                    return BadRequest(new Error
-                    {
-                        Title = "Missing tile",
-                        Message = $"There is no tile with id {id}."
+                    return BadRequest(new Error() {
+                        Title = e.Message
                     });
-                }
+                }                                
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, $"Unknown error while performing {nameof(UpdateUser)} method.");
+                return BadRequest(new UnknownError() { Title = ex.Message });
+            }            
+        }
 
-                currentTile.Users.Clear();
-                currentTile.Users.Add(selectedUser);
+        [HttpPut("{id}")]
+        [Authorize(Roles = UserRoles.SUPERVISOR + "," + UserRoles.EDITOR)]
+        public async Task<ActionResult<string>> Approve(string id, [FromBody] User u)
+        {
+            try
+            {
+                var validationResult = _validationHelper.Validate(ModelState);
+                if (validationResult != null) return BadRequest(validationResult);
 
-                _dbContext.SaveChanges();
+                 
+                
+                try {
+                    DbTile currentTile =
+                    await _dbContext.Tiles
+                        .Include(tile => tile.Approvers)                        
+                        .SingleOrDefaultAsync(x => x.Id == Guid.Parse(id));
 
-                return Ok("User successfully assigned to the tile.");
+                    var user  = await prepareUserForTile(id, u);                    
+                    currentTile.Approvers.Add(user);
+                    _dbContext.SaveChanges();
+
+                    return Ok("Tile approved");
+                } catch (BadHttpRequestException e)
+                {
+                    return BadRequest(new Error() {
+                        Title = e.Message
+                    });
+                }                                
             }
             catch (Exception ex)
             {
                 _logger.LogWarning(ex, $"Unknown error while performing {nameof(UpdateUser)} method.");
                 return BadRequest(new UnknownError() { Title = ex.Message });
             }
         }
+
+        private async Task<ApplicationUser> prepareUserForTile(string id, [FromBody] User u) {
```

Comment:
Nazwy metod z dużej.

Reply:
To [FromBody] jest niepotrzebne mam wrażenie, bo ta metoda nie pobiera niczego z parametrów requestu webowego.

Reply:
Ale wyciągnięcie metody jest ok.

Reply:
poprawione